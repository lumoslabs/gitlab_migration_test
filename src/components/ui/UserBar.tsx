import { css, StyleSheet } from 'aphrodite/no-important';
import commonStyles from '@styles/commonStyles';
import { base } from '@styles/colors';

export interface IUserBarProps {
  clickHandler(e: React.MouseEvent<any>): any;
  profilePicUrl?: string;
  name?: string;
}

const UserBar = (props: IUserBarProps): JSX.Element => {
  return (
    <div className={css([commonStyles.flexRow, styles.footer])}>
      <div className={css([commonStyles.flexRow, styles.clickable])}
        onClick={props.clickHandler}
      >
        <img
          src={props.profilePicUrl || '/assets/guest_avatar.svg'}
          className={css(styles.avatarImg)}
          alt='Avatar'
        />
        <p className={css(styles.nameText)}>
          {props.name || 'Guest' }
        </p>
      </div>
    </div>
  );
};

const styles = StyleSheet.create({
  footer: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    height: '11vmin',
    width: '100%',
    alignItems: 'flex-end',
    marginLeft: '4.58375vmin',
    marginBottom: '4.58375vmin'
  },

  clickable: {
    cursor: 'pointer',
    alignItems: 'flex-end'
  },

  avatarImg: {
    width: '4.58375vmin',
    height: '4.58375vmin',
    borderRadius: '50%'
  },

  nameText: {
    fontSize: '3vmin',
    color: base.lumosBlack,
    fontWeight: 500,
    fontFamily: 'MuseoSans500',
    margin: 0,
    marginLeft: '1.5825vmin'
  },
});

export default UserBar;
