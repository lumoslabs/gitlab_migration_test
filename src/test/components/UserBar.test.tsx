import React from 'react'
import UserBar from '@components/ui/UserBar'
import { render } from '@testing-library/react'

describe('components', () => {
  describe('UserBar', () => {
    it('matches snapshot', () => {
      const mockOnClick = jest.fn();
      const { asFragment } = render(
        <UserBar
          clickHandler={mockOnClick}
        />, {}
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
