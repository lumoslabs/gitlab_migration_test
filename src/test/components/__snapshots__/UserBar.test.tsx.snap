// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components UserBar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flexRow_1r5z645-o_O-footer_smmve6"
  >
    <div
      class="flexRow_1r5z645-o_O-clickable_162zso0"
    >
      <img
        alt="Avatar"
        class="avatarImg_yvir6c"
        src="/assets/guest_avatar.svg"
      />
      <p
        class="nameText_5s9o36"
      >
        Guest
      </p>
    </div>
  </div>
</DocumentFragment>
`;
